src/OpalCompiler-Core/OCASTSemanticAnalyzer.class.st

```diff
@@ -204,7 +204,7 @@ OCASTSemanticAnalyzer >> visitInlinedBlockNode: aBlockNode [
 { #category : #visiting }
 OCASTSemanticAnalyzer >> visitMethodNode: aMethodNode [
 
-	scope := compilationContext scope newMethodScope. 
+	scope := OCMethodScope new outerScope: compilationContext scope.
```

Comment:
I think it should be extracted into CompilationContext>>#newMethodScope. And then maybe it can be merged with #scope method.

Reply:
And it's really just two users of "compilationContext scope".  So it can be transformed into single #newMethodScope method

Reply:
Yes, I will merge this and do another PR with the small fixes